crypto/rsa/rsa_oaep.c

```diff
@@ -244,15 +243,16 @@ int RSA_padding_check_PKCS1_OAEP_mgf1(unsigned char *to, int tlen,
      * should be noted that failure is indistinguishable from normal
      * operation if |tlen| is fixed by protocol.
      */
-    tlen = constant_time_select_int(constant_time_lt(dblen, tlen), dblen, tlen);
+    tlen = constant_time_select_int(constant_time_lt(dblen - mdlen - 1, tlen),
+                                    dblen - mdlen - 1, tlen);
```

Comment:
I think the `good &= constant_time_ge(tlen, mlen);` above should be below this line.

Reply:
And it has a typo in the comment

Reply:
The order is fine.

Reply:
I would be tempted to remove the tlen parameter from the internal API,
and simplify the control flow, based on the knowledge what the maximal plain text length is
num - 2 *mdlen - 2.

Reply:
This is an external API

Reply:
Ouch!